packages/client-sdk-nodejs/src/config/middleware/middleware.ts

```diff
@@ -34,6 +37,548 @@ export class MiddlewareMessage {
     }
     return 0;
   }
+
+  constructorName(): string {
+    return this._grpcMessage.constructor.name;
+  }
+
+  // Note: APIs that use streaming interceptors (e.g. GetBatch and SetBatch)
+  // will not see these debug messages
+  toString(): string {
+    switch (this._grpcMessage.constructor) {
+      case cache.cache_client._GetRequest: {
+        const request = this._grpcMessage as cache.cache_client._GetRequest;
+        return `GetRequest with key "${TEXT_DECODER.decode(
+          request.cache_key
+        )}"`;
+      }
+      case cache.cache_client._SetRequest: {
+        const request = this._grpcMessage as cache.cache_client._SetRequest;
+        return `SetRequest with key "${TEXT_DECODER.decode(
+          request.cache_key
+        )}" and value "${TEXT_DECODER.decode(request.cache_body)}" and ttl ${
```

Comment:
`request.cache_body` could contain megabytes of data. We used a `truncateString()` utility method in the response classes to limit the amount of output from `toString()`. I'm guessing in a situation where someone is using a middleware logger they're going to want to see absolutely everything, but I thought it might be worth discussion.

Reply:
that's a good callout. I think for now let's just leave it the way Anita has it  (logging the whole item) and see if we get feedback.